### PR TITLE
fix xxxPeers interfaces bug

### PR DESCRIPTION
**Pull request by luyikang-York**
1.file writing check
2.file writing format
3.file reading/writing mutex
4.log change
5.return with more details
6. xxxPeers param limit & nodes limit

### CODE REVIEW

File: libp2p/Service.cpp [hunk 1]
```diff
@@ -1162,10 +1172,11 @@ bool Service::erasePeers(std::vector<dev::network::NodeIPEndpoint> const& endpoi
     setStaticNodes(nodes);
```
Comment (ywy2090) [inline libp2p/Service.cpp:1172]: if none endpoint removed , return directly

File: libp2p/Service.cpp [hunk 2]
```diff
@@ -1234,28 +1250,57 @@ bool Service::updatePeersToIni(std::map<dev::network::NodeIPEndpoint, NodeID> co
...
     }
     else
     {
         outfile.flush();
         outfile << fileData;
         outfile.close();
-        SERVICE_LOG(INFO) << LOG_DESC("output data to config file successfully");
+        SERVICE_LOG(INFO) << LOG_DESC("try to output data to config file");
+    }
+    // make sure the output all right
```
Comment (ywy2090) [inline libp2p/Service.cpp:1273]: boost::filesystem::file_size instead

File: libp2p/Service.cpp [hunk 3]
```diff
@@ -1162,10 +1172,11 @@ bool Service::erasePeers(std::vector<dev::network::NodeIPEndpoint> const& endpoi
...
     SERVICE_LOG(INFO) << LOG_DESC("try to update the configfile now");
-    return updatePeersToIni(nodes);
+    return updatePeersToIni(nodes, response);
 }
 
-bool Service::updatePeersToIni(std::map<dev::network::NodeIPEndpoint, NodeID> const& nodes)
+bool Service::updatePeersToIni(
+    std::map<dev::network::NodeIPEndpoint, NodeID> const& nodes, std::string& response)
 {
     std::string tmpdata = "";
```
Comment (ywy2090) [inline libp2p/Service.cpp:1181]: if updatePeersToIni will throw exception, and the method call it handle this exception

**Review by ywy2090 — CHANGES_REQUESTED** [review]

File: libp2p/Service.cpp [hunk 4]
```diff
@@ -44,6 +44,7 @@ using namespace dev::network;
 using namespace dev::stat;
 
 static const uint32_t CHECK_INTERVEL = 10000;
+static const uint32_t MAX_NODES_LIMIT = 300;
```
Comment (ywy2090) [inline libp2p/Service.cpp:47]: 100 is ok

File: libp2p/Service.cpp (hunk 1 above)
Comment (luyikang-York) [reply libp2p/Service.cpp:1172]: ok

File: libp2p/Service.cpp (hunk 4 above)
Comment (luyikang-York) [reply libp2p/Service.cpp:47]: ok

File: libp2p/Service.cpp (hunk 3 above)
Comment (luyikang-York) [reply libp2p/Service.cpp:1181]: ok

**Review by ywy2090 — APPROVED** [review]

**Pull request merged**